profiles/log_signals.py

```diff
@@ -14,34 +16,79 @@ def should_audit(model):
 
 
 def log(action, instance):
-    if settings.AUDIT_LOGGING_ENABLED and should_audit(instance.__class__):
+    def _resolve_role(current_user, profile):
+        if profile.user == current_user:
+            return "OWNER"
+        elif current_user is not None:
+            return "ADMIN"
+        else:
+            return "SYSTEM"
+
+    profile_parts = {
+        "Profile": "base profile",
+        "SensitiveData": "sensitive data",
+    }
+
+    if (
+        settings.AUDIT_LOGGING_ENABLED
+        and should_audit(instance.__class__)
+        and instance.pk
+    ):
         logger = logging.getLogger("audit")
-        # TODO: Finalize log format
-        logger.info(
-            "User %s from service %s %s %s with id %s"
-            % (
-                get_current_user() or "SYSTEM",
-                get_current_service() or "NONE",
-                action,
-                instance.__class__.__name__,
-                str(instance.id),
-            )
-        )
+
+        current_time = datetime.utcnow()
+        current_user = get_current_user()
+        profile = instance.resolve_profile()
+        profile_id = str(profile.pk) if profile else None
+        target_user = profile.user if profile and profile.user else None
+
+        message = {
+            "audit_event": {
+                "origin": "PROFILE-BE",
+                "status": "SUCCESS",
```

Comment:
Maybe at some later point of time we'll have to log events with different a status? Just thinking, nothing that needs to be done in this PR. :+1:

Reply:
Yes that was actually something we already thought. Current implementation doesn't really support for example operations with auth issues 🤔